src/core/net/dnssd_server.cpp

```diff
@@ -102,24 +111,23 @@ void Server::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessag
 
     // ProcessQuery is assumed to always prepare the response DNS message and header properly even in the case of system
     // failures (e.g. no more buffers).
-    ProcessQuery(aMessage, *responseMessage, requestHeader);
-
-    error = mSocket.SendTo(*responseMessage, aMessageInfo);
-
+    ProcessQuery(requestHeader, aMessage, *responseMessage, aMessageInfo);
 exit:
-    FreeMessageOnError(responseMessage, error);
```

Comment:
We need to keep this line because `SuccessOrExit(error = responseMessage->SetLength(sizeof(Header)));` can fail.

Reply:
And, looks like we don't need to alloc the response message here. Instead, it should be better to create the response message when there are really response from SRP server or Discovery Proxy.